### ISSUE TITLE
Unable to fill text until bottom margin

### ISSUE THREAD

**Issue by markpenny** [sillsdev/ptx2pdf#79]
The way that XeTeX fills the page doesn't seem to be working right. If I turn off all footnotes and cross-references, I would expect to be able to fill the page with text right down to the bottom margin (in this case 10mm). However, as I adjust the line spacing in an attempt to fill the page completely, I'm always left with at least 1 blank line at the base of the page (which should be filled with text, but isn't). Click to zoom in on the image below (which shows the margin as 10mm).
![image](https://user-images.githubusercontent.com/12962708/82635198-50b14180-9c1d-11ea-9a59-74dd35054b34.png)

Notice that when I turn on footnotes, they certainly position themselves at very the bottom of the page (within the margins), but there is _often_ an unexplained space above them,
![image](https://user-images.githubusercontent.com/12962708/82635345-a2f26280-9c1d-11ea-9cde-03cffbedf74b.png)

It would be good for the bottom of the text to line up with the bottom of the footnotes (when the line spacing is set correctly).

![image](https://user-images.githubusercontent.com/12962708/82636993-79d3d100-9c21-11ea-9fd2-aa919a549c90.png)

**Comment by mhosken** [comment]
There are many things that can be going on. In the first example, could you try increasing the bottom margin until another line gets eaten. I'm wondering if there is an issue of depth (amount under the baseline) affecting things.

The second example could be due to many things including pulling 4 lines across taking the footnote with it and so not being able to move the lines, or there is a heading involved. I would need to see the whole page. But yes there are times when TeX can be very inscrutable in its line break choice.

**Comment by markpenny** [comment]
I turned OFF the footnotes and X-refs, and found some pages where there was no reason to break, and then I carefully adjusted the line spacing to try to fill the page, but it always left more than one line's worth of space. I'm 99% certain that there is something else between the bottom margin and the bottom of the body text which isn't allowing the text to fill that area..

**Comment by markpenny** [comment]
Thanks @mhosken for fixing this. @davidg-sil - FYI the issue was solved by adding 1 extra depth to the available height (as the last line doesn't need it). We've now got some much prettier output, and will make a lot of other bits work better too.

![image](https://user-images.githubusercontent.com/12962708/82660100-5cb1f900-9c47-11ea-9089-a9a9ae8fcecb.png)

And the last line goes down to the bottom margin (subject to all the usual line spacing settings being right etc.)

![image](https://user-images.githubusercontent.com/12962708/82660209-93880f00-9c47-11ea-8ee0-9f51b6516b2f.png)
